### PR TITLE
PDLP interface

### LINKED ISSUE #1649: Interface to OR-Tools/PDLP
[OR-Tools](https://github.com/google/or-tools) (master branch) now includes a C++ implementation of the PDLP algorithm that we presented at NeurIPS 2021 ([paper](https://proceedings.neurips.cc/paper/2021/hash/a8fbbd3b11424ce032ba813493d95ad7-Abstract.html)). In contrast to the Julia prototype used the paper, the C++ implementation is multithreaded and a bit faster in serial. Our experience is that it's the fastest LP solver among first-order solvers, especially when solving to high precision (we compared with SCS, haven't benchmarked with OSQP). It's not close to being uniformly better than simplex or barrier solvers, but it does outperform them on occasion. We think this solver would be a useful option for CVXPY users.

Could you provide some guidance on how to create a CVXPY interface?

Some relevant technical points:
- PDLP solves LPs and QPs with diagonal objective matrices (only). The underlying algorithm could be generalized to conic problems, but we're unlikely to implement that anytime soon.
- OR-Tools already has a python interface installable through pip and conda. (Though these releases don't include PDLP yet.)
- OR-Tools has its own solver abstraction. It may make sense to create an "or-tools" wrapper rather than a direct "PDLP" wrapper in order to enable users to access other solvers like Glop. Glop is a production-ready simplex-based solver that's used widely within Google.

CC @dapplegate @lperron @Mizux @bodono

### PR DESCRIPTION
**Pull request by mlubin**
Adds an interface to PDLP, using a similar structure to the GLOP interface.

Also switch the Glop interface to use a proto struct rather than a text string for parameters. My impression is that most users won't be familiar with the text format for protos and that it's easier to work with the normal proto object directly.

Notes:
- We were not able to polish off a numpy interface thanks to [pybind11_protobuf](https://github.com/pybind/pybind11_protobuf) being incompatible with cmake. We'll have something for the next release. In the meantime, there will be extra overhead in translating the model between python and C++.
- The apply function is duplicated between the GLOP and the PDLP interface. This code is going to change (and likely diverge) when we have numpy interfaces, so it seemed premature to refactor to share the code.
- I'm not sure of the right way to check that OR-Tools is version 9.3 or better. (Otherwise the code will error out.)
- There's a internal warning printed after each solve, this will go away in the version bump of OR-Tools: `String field 'operations_research.MPSolutionResponse.solver_specific_info' contains invalid UTF-8 data when parsing a protocol buffer. Use the 'bytes' type if you intend to send raw bytes.`

Closes #1649

CC @dapplegate @lperron @Mizux

### CODE REVIEW

**Review by rileyjmurray — CHANGES_REQUESTED** [review]
This mostly looks good. I left comments on how to check the ortools version and about updating the web documentation. You also need to update [this line](https://github.com/cvxpy/cvxpy/blob/master/continuous_integration/install_dependencies.sh#L30) of our CI setup script so that the appropriate version of ortools gets installed. Right now the unit tests are failing on all macOS builds.

File: cvxpy/reductions/solvers/conic_solvers/glop_conif.py [hunk 1]
```diff
@@ -45,8 +45,10 @@ def name(self) -> str:
 
     def import_solver(self) -> None:
         """Imports the solver."""
+        # TODO: Check for OR-Tools version >= 9.3
```
[the remark this answers is not in the document]
Comment (rileyjmurray) [reply cvxpy/reductions/solvers/conic_solvers/glop_conif.py:48]: Note: ``cvxpy.utilities.versioning`` was introduced in a recent commit to master, so you may have to copy that file over to this fork+branch.

[the remark this answers is not in the document]
Comment (mlubin) [reply cvxpy/reductions/solvers/conic_solvers/glop_conif.py:48]: Done

**Review by mlubin — COMMENTED** [review]
> You also need to update [this line](https://github.com/cvxpy/cvxpy/blob/master/continuous_integration/install_dependencies.sh#L30) of our CI setup script so that the appropriate version of ortools gets installed. Right now the unit tests are failing on all macOS builds.

The 9.3 release hasn't yet been posted for macOS. If you prefer I can update install_dependencies.sh to skip installing ortools on macOS, but if I understand correctly the new version checking code should get the tests passing now (because GLOP and PDLP won't show up as installed solvers), and then when 9.3 is available for macOS, the tests will start running it, which is what we want.

**Comment by rileyjmurray** [comment]
@mlubin it seems ``glop_conif.py`` is using ``Version`` but doesn't have ``from cvxpy.utilities.versioning import Version``

**Comment by mlubin** [comment]
> @mlubin it seems glop_conif.py is using Version but doesn't have from cvxpy.utilities.versioning import Version

Just fixed that. I'm confused why tests were passing and this was caught only by the linter.

**Comment by mlubin** [comment]
> Just fixed that. I'm confused why tests were passing and this was caught only by the linter.

Oh, the exception in import_solver would just cause the solver to not be "installed".

**Comment by rileyjmurray** [comment]
> > @mlubin it seems glop_conif.py is using Version but doesn't have from cvxpy.utilities.versioning import Version
> 
> Just fixed that. I'm confused why tests were passing and this was caught only by the linter.

Tests were passing because ``import_solver()`` raised an error, so cvxpy assumed GLOP wasn't installed. Ideally, we should be checking for what kind of error is raised from ``import_solver()`` and have logic to handle the expected case (package not installed) versus unexpected cases (a function being called within ``import_solver()`` that was never imported).

**Review by rileyjmurray — APPROVED** [review]
Looks good! Normally we'd need two project maintainers to approve expanding the API with another solver, so I won't merge just yet. I'll check with Steven and Akshay on Discord to see if they want to review or just approve it as a natural extension of your GLOP PR.

**Review by SteveDiamond — APPROVED** [review]
LGTM

**Comment by Mizux** [comment]
FYI pubind11_protobuf CMake support is tracked here: https://github.com/pybind/pybind11_protobuf/issues/60

**Pull request merged**